go-controller/pkg/ovn/pods.go

```diff
@@ -81,6 +81,27 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 	portInfo, err := oc.logicalPortCache.get(logicalPort)
 	if err != nil {
 		klog.Errorf(err.Error())
+		// If ovnkube-master restarts, it is also possible the Pod's logical switch port
+		// is not readded into the cache. Delete logical switch port anyway.
```

Comment:
@cathy-zhou should we be doing this in syncPods(), since that's what will get called first every time ovnkube-master restarts?

Reply:
the deletion of lsp if pod is deleted during master is down is already taken care of by syncPods(). Here the code is added to delete potential stale entities if master restarts, but addlogicalPort() fails before adding the port to the cache.

Reply:
@cathy-zhou @trozet 

The only thing between creating a Logical Switch Port in OVN and adding the said port to the cache is a call to LSPGet().
```
	// execute all the commands together.
	err = oc.ovnNBClient.Execute(cmds...)
	if err != nil {
		return fmt.Errorf("error while creating logical port %s error: %v",
			portName, err)
	}

	lsp, err = oc.ovnNBClient.LSPGet(portName)
	if err != nil || lsp == nil {
		return fmt.Errorf("failed to get the logical switch port: %s from the ovn client, error: %s", portName, err)
	}

	// Add the pod's logical switch port to the port cache
	portInfo := oc.logicalPortCache.add(logicalSwitch, portName, lsp.UUID, podMac, podIfAddrs)
```

So, the current code is a safeguard for failure of LSPGet() call. I don't think it's anything to do with `ovnkube-master` restart or some such. Also, I don't know why LSPGet() would fail unless there is some OVSDB messages re-ordering in go-ovn library.

Reply:
In downstream, we can see this happens multiple times during CI:
```
I0930 16:46:43.639510       1 pods.go:78] Deleting pod: e2e-resourcequota-priorityclass-2508/testpod-pclass5
E0930 16:46:43.639628       1 pods.go:83] logical port e2e-resourcequota-priorityclass-2508_testpod-pclass5 not found in cache
I0930 16:46:43.687433       1 informer.go:294] Successfully synced 'e2e-resourcequota-priorityclass-2508/testpod-pclass6'
I0930 16:46:43.687520       1 pods.go:78] Deleting pod: e2e-resourcequota-priorityclass-2508/testpod-pclass6
E0930 16:46:43.687592       1 pods.go:83] logical port e2e-resourcequota-priorityclass-2508_testpod-pclass6 not found in cache
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/297/pull-ci-openshift-ovn-kubernetes-master-e2e-ovn-hybrid-step-registry/1311331970055147520/artifacts/e2e-ovn-hybrid-step-registry/gather-extra/pods/openshift-ovn-kubernetes_ovnkube-master-szcbh_ovnkube-master.log

From my perspective, if we get a delete event for a pod that has pod network and we don't have an entry in cache, our cache is invalid and we still need to try to delete it.